lib/sessions.js

```diff
@@ -22,6 +24,94 @@ function assertAlive(session, callback) {
   return true;
 }
 
+// Client session states
+const TxnState = {
```

Comment:
Can we refactor this into something that uses an IIFE to make it less verbose?

I was thinking something like:

```js
let TxnState;
let txnStateTransition;
(() => {
  const NO_TRANSACTION = 'NO_TRANSACTION';
  const STARTING_TRANSACTION = 'STARTING_TRANSACTION';
  const TRANSACTION_IN_PROGRESS: 'TRANSACTION_IN_PROGRESS';
  const TRANSACTION_COMMITTED: 'TRANSACTION_COMMITTED';
  const TRANSACTION_COMMITTED_EMPTY: 'TRANSACTION_COMMITTED_EMPTY';
  const TRANSACTION_ABORTED: 'TRANSACTION_ABORTED';

  const stateMachine = {...};

  TxnState = { NO_TRANSACTION, STARTING_TRANSACTION, TRANSACTIOKN_IN_PROGRESS ... };
  txnStateTransition = function(txnState, nextState) {
  };
})();
```

Reply:
Also, this is adding 114 lines to this file. Can we put it in a separate file?